src/MemoryGraph/graph.cs

```diff
@@ -140,6 +140,10 @@ public virtual NodeType AllocTypeNodeStorage()
         /// The number of references (arcs) in the graph
         /// </summary>
         public int TotalNumberOfReferences { get { return m_totalRefs; } }
+        /// <summary>
+        /// Each segment added to the segmented list used in this class will have a size as specified by this constant.
+        /// </summary>
+        protected static readonly int m_segmentSize = 8_192;
```

Comment:
Make it a const.   I don't think it is interesting to expose this configuration parameter (and we can do so easily later if we ever care to).    Also put a rational for this choice in the comment  (that because T is 8 bytes, it keeps all the segments  ~64K and thus out of the large object heap).  

When you make it const, rename it to be SegmentSize (since it is no longer an member field)